backend/routes/products.js

```diff
@@ -0,0 +1,109 @@
+const express = require('express');
+const router = express.Router();
+const productQueries = require('../queries/products');
+const {handleErrors} = require('./helpers/helpers');
+
+/* GET all product by Brand Id */
+router.get('/:brand_id/all', async (req, res, next) => {
+
+    let brand_id = req.params.brand_id
+
+    try {
+        let productByBrand = await productQueries.getProductsByBrand(brand_id)
+        res.status(200).json({
+            message: `Success retrieved all products for brand with id #${brand_id}`,
+            payload: productByBrand
+        })
+
+    } catch (err) {
+        handleErrors(res, err)
+    }
+})
+
+/* GET all product by type Id */
+router.get('/:type_id', async (req, res) =>{
+    let type_id = req.params.type_id
+
+    try{
+        let productByType = await productQueries.getProductByType(type_id)
+        res.status(200).json({
+            message: `Success, retrieved all products by type with id #${type_id}`, 
+            payload: productByType
+        })
+
+    } catch(err) {
+        handleErrors(res, err)
+    }
+})
+
+/* GET all product by textile Id */
+router.get('/:textile_id', async(req, res) => {
+    let material_id = req.params.textile_id
+
+    try{
+        let productByMaterial = await productQueries.getProductByMaterial(material_id)
+        res.status(200).json({
+            message: `Success, retrieved all products by material with id #${material_id}`, 
+            payload: productByMaterial
+        })
+
+    } catch (error) {
+        handleErrors(res, error)
+    }
+})
+
+
+/* POST new product*/
+router.post('/add/:brand_id', async(req, res) =>{
+
+    let brand_id = req.params.brand_id
+    let type_id = req.body.type_id
+    let name = req.body.name
+    let default_pic = req.body.default_pic
```

Comment:
We will have a different approach to have the image, most likely using Multer the same way we did the our second group project